dev/com.ibm.ws.logging.osgi/src/com/ibm/ws/logging/internal/osgi/stackjoiner/ThrowableProxyActivator.java

```diff
@@ -0,0 +1,395 @@
+import com.ibm.ws.logging.internal.osgi.stackjoiner.bci.ThrowableClassFileTransformer;
+import com.ibm.ws.logging.internal.osgi.stackjoiner.boot.templates.ThrowableProxy;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.commons.ClassRemapper;
+import org.objectweb.asm.commons.SimpleRemapper;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Bundle;
+
+public class ThrowableProxyActivator {
+	
+	private ThrowableInfo throwableInfo;
+	private Instrumentation inst;
+	private BundleContext bundleContext;
+	
+	/**
+     * The target package for the boot loader delegated classes.
+     */
+    public final static String BOOT_DELEGATED_PACKAGE = "com.ibm.ws.boot.delegated.logging";
+    
+    /**
+     * The bundle entry path prefix to the template classes.
+     */
+    final static String TEMPLATE_CLASSES_PATH = ThrowableProxy.class.getPackage().getName().replaceAll("\\.", "/");
+    
+    /**
+     * The name of the {@link ThrowableProxy} class that needs to be
+     * made available on the bootstrap class loader.
+     */
+    public final static String THROWABLE_PROXY_CLASS_NAME = BOOT_DELEGATED_PACKAGE + "." + ThrowableProxy.class.getSimpleName();
+
+    /**
+     * The internal name of the {@link ThrowableProxy} class.
+     */
+    public final static String THROWABLE_PROXY_CLASS_INTERNAL_NAME = THROWABLE_PROXY_CLASS_NAME.replaceAll("\\.", "/");
+    
+    /**
+     * The name of the field in generated classes that will contain the
+     * generating bundle version.
+     */
+    final static String VERSION_FIELD_NAME = "BUNDLE_VERSION";
+
+    /**
+     * The proxy jar manifest header that indicates the version of the bundle
+     * that created the jar.
+     */
+    final static String LOGGING_VERSION_MANIFEST_HEADER = "Liberty-Logging-Osgi-Bundle-Version";
+    
+	
+    public ThrowableProxyActivator(Instrumentation inst, BundleContext bundleContext) {
+    	this.inst = inst;
+    	this.bundleContext = bundleContext;
+    }
+    
+    /**
+     * Activation callback from the Declarative Services runtime where the
+     * component is ready for activation.
+     *
+     * @param bundleContext the bundleContext
+     */
+    public void activate() throws Exception {    
+        // Store a reference to the printStackTraceOverride method from BaseTraceService
+        throwableInfo = new ThrowableInfo(inst);
+		
+        if (throwableInfo.isInitialized()) {
+        	String runtimeVersion = getRuntimeClassVersion();
+            if (runtimeVersion != null && !runtimeVersion.equals(getCurrentVersion())) {
+                // TODO: Use a compatibility check instead
+                throw new IllegalStateException("Incompatible proxy code (version " + runtimeVersion + ")");
+            }
```

Comment:
I can't see this issue occurring unless the Throwable Proxy jar is being modified somewhere else.. 
I think it may be best to just add an additional condition that will check if a new jar file needs to be created (as shown in [f79ffd1](https://github.com/OpenLiberty/open-liberty/pull/16276/commits/f79ffd163f2464ddfa7715700424ffc110c2c1dd)).